compiler/rustc_resolve/src/diagnostics.rs

```diff
@@ -1421,14 +1419,26 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
             "",
         );
 
+        if macro_kind == MacroKind::Bang
+            && ident.name == sym::macro_rules
+            && let source_map = self.tcx.sess.source_map()
+            && let Ok(next_source) = source_map.span_to_next_source(ident.span)
+            && next_source.starts_with("!")
```

Comment:
This condition is actually also redundant.
`macro_kind == MacroKind::Bang` means that the next token is `!` by definition (maybe after some whitespace, but that's fine).

Reply:
> maybe after some whitespace, but that's fine

```rs
// amazing it to work normally :)
macro_rules         !  something { () => {}}
```